packages/core/src/error.ts

```diff
@@ -13,14 +15,45 @@ export class ApplicationError extends Error {
   constructor(
     public readonly type: string,
     public readonly status: number,
-    public readonly title: string
+    public readonly title: string,
+    public readonly detail?: string
   ) {
-    super(title)
+    super(detail || title)
   }
 }
 
 export class IllegalStateError extends ApplicationError {
   constructor(description: string) {
-    super("/problem/illegal-state", 500, `Illegal state reached: ${description}`)
+    const { type, code, title } = serverErrorResponse.InternalServerError
```

Comment:
I would much rather overload the constructor of ApplicationError than have to destructure every single time

Reply:
I had not thought about that, I think I like that better too. I'll try it out.